src/styles/components/_calendar.scss

```
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  width: $container-width;
+  margin: auto;
+  gap: $gap;
+  padding: $padding;
+
+  @for $i from 28 through 31 {
+    $decimer: 31 - $i;
+    &--month-length-#{$i} .calendar__day:nth-last-child(-n + #{$decimer}) {
+      display: none;
+    }
+  }
+
+  @each $day, $margin in $week-days {
+    &--start-day-#{$day} .calendar__day--1 {
+      margin-left: $day-size * $margin + $gap * $margin;
+    }
+  }
+
+  &__day {
+    @include day-size($day-size);
+    position: relative;
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translate(0, -20px);
+    }
+  }
+
+  @for $i from 1 through 31 {
+    &__day--#{$i}::before {
```

```suggestion
    &__day--#{$i}::before {
```
You have selector &__day higher, move it there